packages/@aws-cdk/aws-cloud9/lib/environment.ts

```diff
@@ -0,0 +1,134 @@
+   * the Amazon EC2 instance.
+   *
+   * @default - all public subnets of the VPC are selected.
+   */
+  readonly subnetSelection?: ec2.SubnetSelection;
+
+  /**
+   * The VPC that AWS Cloud9 will use to communicate with the Amazon Elastic Compute Cloud (Amazon EC2) instance.
+   *
+   */
+  readonly vpc: ec2.IVpc;
+
+  /**
+   * Name of the environment
+   *
+   * @default - automatically generated name
+   */
+  readonly environmentEc2Name?: string;
+
+  /**
+   * Description of the environment
+   *
+   * @default - no description
+   */
+  readonly description?: string;
+}
+
+/**
+ * A Cloud9 Environment with Amazon EC2
+ */
+export class EnvironmentEC2 extends cdk.Resource implements IEnvironmentEC2 {
+  /**
+   * import from EnvironmentEC2Name
+   */
+  public static fromEnvironmentEC2Name(scope: cdk.Construct, id: string, environmentEC2Name: string): IEnvironmentEC2 {
+    class Import extends cdk.Resource {
+      public environmentEc2Name = environmentEC2Name;
+      public environmentEc2Arn = cdk.Stack.of(this).formatArn({
+        service: 'cloud9',
+        resource: 'cloud9:*:environmentEc2Name',
+        resourceName: this.environmentEc2Name
+      });
+    }
+    return new Import(scope, id);
+  }
+
+  /**
+   * The environment name of this Cloud9 environment
+   */
+  public readonly environmentEc2Name: string;
+
+  /**
+   * The environment ARN of this Cloud9 environment
+   */
+  public readonly environmentEc2Arn: string;
+
+  /**
+   * The environment ID of this Cloud9 environment
+   */
+  public readonly environmentId: string;
+
+  /**
+   * VPC ID
+   */
+  public readonly vpc: ec2.IVpc;
+
+  /**
+   * The complete IDE URL of this Cloud9 environment
+   */
+  public readonly ideUrl: string;
+
+  constructor(scope: cdk.Construct, id: string, props: EnvironmentEC2Props) {
+    super(scope, id);
+    this.vpc = props.vpc;
+    // if (props.subnetSelection && this.vpc.selectSubnets(props.subnetSelection).subnetIds.length === 0) {
+    //   throw new Error('no subnet found with provided subnetSelection');
+    // }
+
+    // if (!props.subnetSelection && this.vpc.selectSubnets({ subnetType: ec2.SubnetType.PUBLIC } ).subnetIds.length === 0) {
+    //   throw new Error('no subnetSelection specified and no public subnet found in the vpc, please specify subnetSelection.');
+    // }
+
+    if (!props.subnetSelection && this.vpc.publicSubnets.length === 0) {
+      throw new Error('no subnetSelection specified and no public subnet found in the vpc, please specify subnetSelection');
+    }
+    const c9env = new CfnEnvironmentEC2(this, 'Resource', {
+      name: props.environmentEc2Name,
+      description: props.description,
+      instanceType: props.instanceType?.toString() ?? ec2.InstanceType.of(ec2.InstanceClass.BURSTABLE2, ec2.InstanceSize.MICRO).toString(),
+      subnetId: props.subnetSelection ? this.vpc.selectSubnets(props.subnetSelection).subnetIds[0] :
```

Comment:
None of this is required. Just do:

```ts
subnetId: props.vpc.selectSubnets(props.subnetSelection).subnetIds[0],
```

Reply:
Yes, but `props.subnetSelection` might be undefined as its optional property and in this case we just select the first public subnet instead. What do you think?

Reply:
It's not a problem that `props.subnetSelection` might be undefined. [`selectSubnets`'s first argument is optional](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-ec2.Vpc.html#select-wbr-subnetsselection).

Reply:
OK. Will fix it 😉

Reply:
@skinny85 It turned out if `props.subnetSelection` is undefined, the private subnet will be selected and the cloud9 Ec2 can not be accessed from public internet.

I would like the default behavior will select public subnet when `props.subnetSelection` is undefined unless the  it's explicitly specified. Thoughts?

Reply:
OK. So something like:

```ts
const vpcSubnets = props.subnetSelection ?? { type: ec2.SubnetType.PUBLIC };

// ...

  subnetId: props.vpc.selectSubnets(vpcSubnets).subnetIds[0] :
```

Reply:
Thanks. Fixed.